### PR TITLE
feat: configure cookie to be fetched from exact domain

### PR DESCRIPTION
**Pull request by MoumitaM**
## PR Description

Please include a summary of the change along with the relevant motivation and context.

## Linear task (optional)

Linear task link

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by saikumarrs — DISMISSED** [review]

**Comment by bardisg** [comment]
@MoumitaM does this cover both not adding cookies in top domain and not reading values from top domain if the new flag is true?

**Comment by MoumitaM** [comment]
> @MoumitaM does this cover both not adding cookies in top domain and not reading values from top domain if the new flag is true?
With this feature, cookies will only be set from the current exact domain and read from that domain only. Say cookies set in testwebsite.com can only be accessed by testwebsite.com not by any other sub domains. The same is true for sub domains also.

File: packages/analytics-js-common/src/types/LoadOptions.ts [hunk 1]
```diff
@@ -145,6 +145,7 @@ export type LoadOptions = {
   // transport mechanism to be used for sending batched requests
   transportMode?: EventsTransportMode; // Unused for now. This will deprecate the useBeacon and beaconQueueOptions
   consentManagement?: ConsentManagementOptions;
+  sameDomainCookieOnly?: boolean;
```
Comment (saikumarrs) [inline packages/analytics-js-common/src/types/LoadOptions.ts:148]:
```suggestion
  sameDomainCookiesOnly?: boolean;
```

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: packages/analytics-js-common/src/types/LoadOptions.ts (hunk 1 above)
Comment (MoumitaM) [reply packages/analytics-js-common/src/types/LoadOptions.ts]: Done

**Review by saikumarrs — APPROVED** [review]

**Review by bardisg — APPROVED** [review]

**Pull request merged**